src/Common/getNumberOfCPUCoresToUse.cpp

```diff
@@ -189,9 +189,9 @@ unsigned getNumberOfPhysicalCPUCoresImpl()
 
 }
 
-unsigned getNumberOfPhysicalCPUCores()
+unsigned getNumberOfCPUCoresToUse()
 {
     /// Calculate once.
-    static auto cores = getNumberOfPhysicalCPUCoresImpl();
+    static const unsigned cores = getNumberOfCPUCoresToUseImpl();
```

Comment:
LGTM, but I wonder if this value should be refreshed at some point when using cgroups (as we do for memory), although that's something for a different PR if needed.

Reply:
i think ideally yes, but i could also imagine it being not easy to implement, because then you have to update (most likely recreate) all the pools and other objects that were constructed with number of cores among the arguments.

Reply:
We could register callbacks and the limitation that nº of threads can only go up (like we do for some config parameters) but probably not worth it if nobody has asked for it.